api/src/main/java/io/minio/MinioClient.java

```diff
@@ -2774,27 +2901,198 @@ public void putObject(String bucketName, String objectName, InputStream stream,
+   *   }
+   *   ByteArrayInputStream bais = new ByteArrayInputStream(builder.toString().getBytes("UTF-8"));
+   * 
+   *   // Generate symmetric 256 bit AES key.
+   *   KeyGenerator symKeyGenerator = KeyGenerator.getInstance("AES");
+   *   symKeyGenerator.init(256);
+   *   SecretKey symKey = symKeyGenerator.generateKey();
+   *   EncryptionMaterials encMaterials = new EncryptionMaterials(symKey);
+   * 
+   *   // create object
+   *   minioClient.putObject("my-bucketname", "my-objectname", bais, bais.available(), "application/octet-stream",
+   *       encryptionMaterials);
+   *   bais.close();
+   *   System.out.println("my-bucketname is uploaded successfully");
+   * }
+   * </pre>
+   *
+   * @param bucketName
+   *          Bucket name.
+   * @param objectName
+   *          Object name to create in the bucket.
+   * @param stream
+   *          stream to upload.
+   * @param size
+   *          Size of all the data that will be uploaded.
+   * @param contentType
+   *          Content type of the stream.
+   * @param encryptionMaterials
+   *          Encryption metadata details and operations.
+   *
+   * @throws InvalidBucketNameException
+   *           upon invalid bucket name is given
+   * @throws NoResponseException
+   *           upon no response from server
+   * @throws IOException
+   *           upon connection error
+   * @throws XmlPullParserException
+   *           upon parsing response xml
+   * @throws ErrorResponseException
+   *           upon unsuccessful execution
+   * @throws InternalException
+   *           upon internal library error
+   * @throws InvalidAlgorithmParameterException
+   *           upon wrong encryption algorithm used
+   * @throws BadPaddingException
+   *           upon incorrect padding in a block
+   * @throws IllegalBlockSizeException
+   *           upon incorrect block
+   * @throws NoSuchPaddingException
+   *           upon wrong padding type specified
+   *
+   * @see #putObject(String bucketName, String objectName, String fileName, String contentType)
+   * @see #putObject(String bucketName, String objectName, String fileName)
+   * @see #putObject(String bucketName, String objectName, InputStream stream, long size, String contentType)
+   *
+   */
+  public void putObject(String bucketName, String objectName, InputStream stream, long size, String contentType,
+      EncryptionMaterials encryptionMaterials) throws InvalidBucketNameException, NoSuchAlgorithmException,
+      InsufficientDataException, IOException, InvalidKeyException, NoResponseException, XmlPullParserException,
+      ErrorResponseException, InternalException, InvalidArgumentException, NoSuchPaddingException,
+      IllegalBlockSizeException, BadPaddingException, InvalidAlgorithmParameterException {
+
+    if (encryptionMaterials == null) {
+      throw new InvalidArgumentException("empty encryption properties not allowed");
+    }
+
+    // Generate symmetric 256 bit AES key for data encryption
```

Comment:
`minio-go` may have lack of support or limitation to do so.  But its clean to do in java with method overloading.  Its not a stopper I guess.